From bfb6aa74d52c369d3bc136d8d5e869ec61a748b6 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Wed, 13 Nov 2013 17:32:59 -0500
Subject: [PATCH] Make --update-cache work better

Due to an oversight, combining the --update-cache option with
explicitly specified modules was leading to an attempt to
open the --update-cache.so module - with predictably disappointing
results. Make this work as intended, and as described
in the man page.
---
 gdk-pixbuf/queryloaders.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdk-pixbuf/queryloaders.c b/gdk-pixbuf/queryloaders.c
index f1b22ce..d23654e 100644
--- a/gdk-pixbuf/queryloaders.c
+++ b/gdk-pixbuf/queryloaders.c
@@ -374,7 +374,7 @@ int main (int argc, char **argv)
         else {
                 char *cwd = g_get_current_dir ();
 
-                for (i = 1; i < argc; i++) {
+                for (i = first_file; i < argc; i++) {
                         char *infilename = argv[i];
 #ifdef G_OS_WIN32
                         infilename = g_locale_to_utf8 (infilename,
-- 
1.8.4.2

